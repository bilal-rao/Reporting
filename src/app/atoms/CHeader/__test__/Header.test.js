import { render, screen, cleanup, waitFor } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import '@testing-library/jest-dom';

//Component
import Header from '../index';
import { Provider } from 'react-redux';

const mockStore = configureStore([]);


afterEach(cleanup);

describe('Header Component', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      global: {
        menu: false,
      },
    });
  });

  it('Render Header component with heading passed as Props', async () => {
    waitFor(() => {
      render(
        <Provider store={store}>
          <Header title="bilal" />
        </Provider>,
      );
      const titleElement = screen.getByText('/bilal/i');
      expect(titleElement).toBeInTheDocument();
    });
  });

  it('Test Header component against snapshot', async () => {
    const { container } = render(
      <Provider store={store}>
        <Header title="bilal" />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
