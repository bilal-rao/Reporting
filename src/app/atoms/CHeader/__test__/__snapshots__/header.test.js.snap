// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Test Header component against snapshot 1`] = `
<div>
  <div
    class="ant-card c-header"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="ant-row ant-row-middle"
        style="margin: -7.5px -12px -7.5px -12px;"
      >
        <div
          class="ant-col ant-col-12 ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-12 ant-col-xl-12"
          style="padding: 7.5px 12px 7.5px 12px;"
        >
          <div
            class="ant-space ant-space-horizontal ant-space-align-center w-100 text-center pl-0"
          >
            <div
              class="ant-space-item"
              style="margin-right: 20px;"
            >
              <div
                class="ant-image"
              >
                <img
                  alt="logo"
                  class="ant-image-img logo"
                  src="[object Object]"
                />
              </div>
            </div>
            <div
              class="ant-space-item"
            >
              <span
                aria-label="align-left"
                class="anticon anticon-align-left"
                role="img"
                style="font-size: 18px; padding-top: 10px; color: cornflowerblue;"
                tabindex="-1"
              >
                <svg
                  aria-hidden="true"
                  data-icon="align-left"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M120 230h496c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm0 424h496c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm784 140H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0-424H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-col ant-col-12 text-right ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-12 ant-col-xl-12"
          style="padding: 7.5px 12px 7.5px 12px;"
        >
          <div
            class="ant-space ant-space-horizontal ant-space-align-center pl-0"
          >
            <div
              class="ant-space-item"
            >
              <div
                class="ant-space ant-space-horizontal ant-space-align-center ant-dropdown-trigger userDropdown"
              >
                <div
                  class="ant-space-item"
                  style="margin-right: 8px;"
                >
                  <span
                    class="ant-avatar ant-avatar-square user-image"
                    style="width: 35px; height: 35px; line-height: 35px; font-size: 18px;"
                  >
                    <span
                      class="ant-avatar-string"
                      style="opacity: 0;"
                    >
                      B
                    </span>
                  </span>
                </div>
                <div
                  class="ant-space-item"
                >
                  <span
                    class="ant-typography user-text"
                  >
                    bilal
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
